apps/discovery_api/test/unit/discovery_api/event_handler_test.exs

```diff
@@ -65,4 +68,30 @@ defmodule DiscoveryApi.EventHandlerTest do
       assert_called(DataJsonService.delete_data_json())
     end
   end
+
+  describe "handle_event/1 #{dataset_delete()}" do
+    setup do
+      %{dataset: TDG.create_dataset(%{id: Faker.UUID.v4()})}
+    end
+
+    test "should delete the dataset and return ok when dataset:delete is called", %{dataset: dataset} do
+      expect(DiscoveryApi.RecommendationEngine.delete(dataset.id), return: :ok)
+      expect(StatsCalculator.delete_completeness(dataset.id), return: :ok)
+      expect(ResponseCache.invalidate(), return: {:ok, true})
+      expect(SystemNameCache.delete(dataset.technical.orgName, dataset.technical.dataName), return: {:ok, true})
+      expect(Model.delete(dataset.id), return: :ok)
+      Brook.Event.process(:discovery_api, Brook.Event.new(type: dataset_delete(), data: dataset, author: :author))
+      assert {:ok, []} == DataJsonService.delete_data_json()
+    end
+
+    test "should return ok if it throws error when dataset:delete is called", %{dataset: dataset} do
+      expect(DiscoveryApi.RecommendationEngine.delete(dataset.id), return: {:error, "ERR value is not an integer or out of range"})
+      expect(StatsCalculator.delete_completeness(dataset.id), return: {:error, "ERR value is not an integer or out of range"})
+      expect(ResponseCache.invalidate(), return: {:ok, true})
+      expect(SystemNameCache.delete(dataset.technical.orgName, dataset.technical.dataName), return: {:ok, true})
+      expect(Model.delete(dataset.id), return: {:error, "ERR value is not an integer or out of range"})
+      Brook.Event.process(:discovery_api, Brook.Event.new(type: dataset_delete(), data: dataset, author: :author))
+      assert {:ok, []} == DataJsonService.delete_data_json()
```

Comment:
At least one of these above should actually mock out raising an error.  Otherwise you're not really testing the rescue.  Try something like this:

```
allow(Model.delete(dataset.id), exec: fn _ -> raise {:error, "error"} end)
```